Sprintfinity3902/Dungeon/Dungeon.cs

```diff
@@ -133,7 +135,9 @@ public IRoom GetCurrentRoom()
 
         public void SetCurrentRoom(int id)
         {
-            NextId = id;
+            CurrentRoom.garbage.Clear();
+            CurrentRoom = GetById(id);
+            SetLinkPosition();
```

Comment:
This is the source of the door bug! Link's position shouldn't be set by SetCurrentRoom, if you comment out this line it should work fine. (Brad helped me hunt this down.)

Reply:
The second half to this bug is mentioned here: https://github.com/kylegessel/sprintfinity3902/pull/63#discussion_r602934422